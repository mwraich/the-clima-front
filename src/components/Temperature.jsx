import React from "react";
import PropTypes from "prop-types";
import {
  TemperatureBox, // wraps the content inside the Temperature component
  LargeText, // The top larger text inside this box
  SmallText // The bottom smaller text inside this box
} from "../elements/box";

/* props:
- temp (number)
- city (string)
*/

export default class Temperature extends React.Component
{
  static propTypes = {
    temp: PropTypes.number.isRequired,
    city: PropTypes.string.isRequired
  };

  render() {
    return (
      <TemperatureBox>
        <LargeText> {this.props.temp}&deg;</LargeText>
        <SmallText> {this.props.city} </SmallText>
      </TemperatureBox>
    );
  }
}
