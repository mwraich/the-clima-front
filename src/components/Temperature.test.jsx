import Temperature from "./Temperature";

it('renders correctly', () => {
  const wrapper = render(
    <Temperature city="Toronto" temp={10} toggleForecast={
        () => {}}/>
    );
    expect(wrapper).toMatchSnapshot();
});
