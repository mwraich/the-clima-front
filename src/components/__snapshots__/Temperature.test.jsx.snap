// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="sc-bwzfXH ilYHYa sc-bdVaJa dQdFv"
>
  <span
    class="sc-ifAKCX XBTtJ"
  >
     10Â°
  </span>
  <span
    class="sc-EHOje exztiK"
  >
     Toronto 
  </span>
</div>
`;
